src/core/sgrubberband.cpp

```diff
@@ -73,39 +78,36 @@ QSGGeometryNode *SGRubberband::createLineGeometry( const QVector<QgsPoint> &poin
 
 QSGGeometryNode *SGRubberband::createPolygonGeometry( const QVector<QgsPoint> &points )
 {
-  double *coordinates_out;
-  int *tris_out;
-  int nverts, ntris;
-
-  double *vertices_in = ( double * ) malloc( points.size() * 2 * sizeof( double ) );
-  const double *contours_array[] = { vertices_in, vertices_in + points.size() * 2 };
-  int i = 0;
-
-  for ( const QgsPoint &pt : points )
-  {
-    vertices_in[i++] = pt.x();
-    vertices_in[i++] = pt.y();
-  }
-
-  tessellate( &coordinates_out, &nverts,
-              &tris_out, &ntris,
-              contours_array, contours_array + 2 );
+  QgsPolygon *polygon = new QgsPolygon( new QgsLineString( points ) );
+  QgsTessellator t( 0, 0, false, false, false, true );
+  if ( points.size() > 2 )
+    t.addPolygon( *polygon, 0 );
```

Comment:
```suggestion
  QgsPolygon polygon = QgsPolygon( new QgsLineString( points ) );
  QgsTessellator t( 0, 0, false, false, false, true );
  if ( points.size() > 2 )
    t.addPolygon( polygon, 0 );
```

This avoids an unnecessary copy and doesn't leak with points.size() <= 2